### DISCUSSION TITLE
Can't create large matrix (more than 173_589 elements)

### DISCUSSION THREAD

**Discussion by EnvOut** [dimforge/nalgebra#1307] [Q&A] [answered]
Hello guys
I try to create vector with more than 173_589 elements:

```
const M: usize = 173_590;
let v1: OMatrix<f32, Const<M>, U1> = SVector::new_random();
```
    
This throws error in compile time
 ```
thread 'main' has overflowed its stack
fatal runtime error: stack overflow

 ```
    
How to create vector with millions (1_000_000) elements?

**Comment by Ralith** [comment, marked as answer]
Use a `DVector` instead. `SVector`s store elements on the stack, which has limited space. Even if it fit, you wouldn't want to use a large `SVector` in most cases because it will be disproportionately expensive to move around.
